src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/DynamicTableViewCell.swift

```diff
@@ -62,12 +83,18 @@ extension DynamicCell {
 }
 
 extension DynamicCell {
-	static func dynamicType(text: String, size: CGFloat = 17, weight: UIFont.Weight = .regular, style: UIFont.TextStyle = .body, color: UIColor? = nil, accessibilityIdentifier: String? = nil, accessibilityTraits: UIAccessibilityTraits = .staticText, configure: CellConfigurator? = nil) -> Self {
-		.identifier(CellReuseIdentifier.dynamicTypeText, action: .none, accessoryAction: .none) { viewController, cell, indexPath in
-			guard let cell = cell as? DynamicTypeTableViewCell else { return }
+	static func dynamicType(text: String, cellStyle: TextCellStyle = .label, size: CGFloat = 17, weight: UIFont.Weight = .regular, style: UIFont.TextStyle = .body, color: UIColor? = nil, accessibilityIdentifier: String? = nil, accessibilityTraits: UIAccessibilityTraits = .staticText, configure: CellConfigurator? = nil) -> Self {
+		.identifier(cellStyle.reuseIdentifier, action: .none, accessoryAction: .none) { viewController, cell, indexPath in
+			guard let cell = cell as? DynamicTableViewTextCell else { return }
 			cell.configureDynamicType(size: size, weight: weight, style: style)
 			cell.configure(text: text, color: color)
 			cell.configureAccessibility(label: text, identifier: accessibilityIdentifier, traits: accessibilityTraits)
+
+			if case .textView(let dataDetectorTypes) = cellStyle,
+				let cell = cell as? DynamicTableViewTextViewCell {
+				cell.dataDetectorTypes = dataDetectorTypes
+			}
```

Comment:
This looks very dirty, but to me seemed the easiest way without some bigger refactoring.

Reply:
Looks perfectly fine for me.